### PR TITLE
Added scheduler for dutch

### PR DESCRIPTION
**Pull request by VDigitall**
This change is [Reviewable](https://reviewable.io/reviews/openprocurement/openprocurement.chronograph/10)

### CODE REVIEW

File: openprocurement/chronograph/scheduler.py [hunk 1]
```diff
@@ -64,41 +65,67 @@ def delete_holiday(db, day):
...
-    end = start + bids * BIDDER_TIME + SERVICE_TIME + MIN_PAUSE
-    seconds = (end - TZ.localize(datetime.combine(end, WORKING_DAY_START))).seconds
+def calc_auction_end_time(bids, start, classic_auction=True):
+    if classic_auction:
+        end = start + bids * BIDDER_TIME + SERVICE_TIME + MIN_PAUSE
+        seconds = (
+            end - TZ.localize(datetime.combine(end, WORKING_DAY_START))
+        ).seconds
+    else:
+        end = start + timedelta(minutes=480)
```
Comment (kroman0) [inline openprocurement/chronograph/scheduler.py]: 480 додай як константу

File: openprocurement/chronograph/scheduler.py [hunk 2]
```diff
@@ -115,49 +142,64 @@ def find_free_slot(plan):
 def planning_auction(auction, start, db, quick=False, lot_id=None):
     tid = auction.get('id', '')
     mode = auction.get('mode', '')
-    calendar = get_calendar(db)
-    streams = get_streams(db)
+    classic_auction = False \
+        if auction.get('procurementMethodType') == 'dgfInsider' \
```
Comment (kroman0) [inline openprocurement/chronograph/scheduler.py]: І тут константу з переліком.

File: openprocurement/chronograph/scheduler.py [hunk 3]
```diff
@@ -64,41 +65,67 @@ def delete_holiday(db, day):
 
 
 def get_streams(db, streams_id=STREAMS_ID):
```
Comment (kroman0) [inline openprocurement/chronograph/scheduler.py]: Додай classic_auction=True і віддавай лише необхідні дані

File: openprocurement/chronograph/design.py [hunk 4]
```diff
@@ -12,14 +12,23 @@ def sync_design(db):
...
+                            emit([x[0], null],
+                                 doc._id.split('_')[1] + 'T' + t);
+                        }
                     }
                 }
             }
+            if (i.indexOf('dutch_st') == 0) {
+                for (var aid in doc[i]) {
+                    emit([doc[i][aid], null],
+                          doc._id.split('_')[1] + 'T09:00:00');
```
Comment (kroman0) [inline openprocurement/chronograph/design.py]: Це б з константи взяти.

Comment (kroman0) [reply openprocurement/chronograph/design.py]: INSIDER_WORKING_DAY_START ніби 09:30?

File: openprocurement/chronograph/scheduler.py [hunk 5]
```diff
@@ -63,42 +72,70 @@ def delete_holiday(db, day):
         db.save(calendar)
 
 
-def get_streams(db, streams_id=STREAMS_ID):
-    return db.get(streams_id, {'_id': streams_id, 'streams': 10}).get('streams')
+def get_streams(db, streams_id=STREAMS_ID, classic_auction=True):
+        streams = db.get(streams_id, DEFAULT_STREAMS_DOC)
+        if classic_auction:
+            return streams['streams']
```
Comment (kroman0) [inline openprocurement/chronograph/scheduler.py]: ['streams'] => .get('streams', DEFAULT_STREAMS_DOC['streams'])

File: openprocurement/chronograph/scheduler.py [hunk 6]
```diff
@@ -63,42 +72,70 @@ def delete_holiday(db, day):
...
+        else:
+            return streams['dutch_streams']
```
Comment (kroman0) [inline openprocurement/chronograph/scheduler.py]: І тут.

File: openprocurement/chronograph/scheduler.py [hunk 7]
```diff
@@ -63,42 +72,70 @@ def delete_holiday(db, day):
...
+        plan[stream_id] = stream
+    else:
+        dutch_streams = plan.get('dutch_streams', [])
+        dutch_streams.append(auction_id)
+        plan['dutch_streams'] = dutch_streams
     db.save(plan)
 
 
-def calc_auction_end_time(bids, start):
+def calc_auction_end_time(bids, start, classic_auction=True):
```
Comment (kroman0) [inline openprocurement/chronograph/scheduler.py]: Тут не потрібно змін.

File: openprocurement/chronograph/design.py [hunk 8]
```diff
@@ -12,14 +12,23 @@ def sync_design(db):
...
+                        if (x.length == 2) {
+                            emit(x, doc._id.split('_')[1] + 'T' + t);
+                        } else {
+                            emit([x[0], null],
+                                 doc._id.split('_')[1] + 'T' + t);
+                        }
                     }
                 }
             }
+            if (i.indexOf('dutch_st') == 0) {
```
Comment (kroman0) [inline openprocurement/chronograph/design.py]: Чому dutch_st а не dutch_streams? В цьому місці підійде пряме порівняння.

**Comment by VDigitall** [comment]
@kroman0, @Leits виправив.

File: openprocurement/chronograph/scheduler.py (hunk 5 above)
Comment (kroman0) [reply openprocurement/chronograph/scheduler.py]: ping

File: openprocurement/chronograph/scheduler.py (hunk 6 above)
Comment (kroman0) [reply openprocurement/chronograph/scheduler.py]: pong

File: openprocurement/chronograph/scheduler.py [hunk 9]
```diff
@@ -115,49 +140,56 @@ def find_free_slot(plan):
...
     if quick:
         quick_start = calc_auction_end_time(0, start)
         return (quick_start, 0, skipped_days)
+    calendar = get_calendar(db)
+    streams = get_streams(db, classic_auction=classic_auction)
     start += timedelta(hours=1)
-    if start.time() < WORKING_DAY_START:
-        nextDate = start.date()
-    else:
+    if classic_auction and start.time() > WORKING_DAY_START:
```
Comment (kroman0) [inline openprocurement/chronograph/scheduler.py]: Яке значення shouldStartAfter для dgfInsider?
@yshalenyk @Leits

**Comment by kroman0** [comment]
Reviewed 3 of 5 files at r1, 3 of 7 files at r4, 3 of 5 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[openprocurement/chronograph/design.py, line 26 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KtQXGe3658uBbFaYD7z-r4-26:-KuUNH8Fr3UbG78mw3fp:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/design.py#L26)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

Чому dutch_st а не dutch_streams? В цьому місці підійде пряме порівняння.
</blockquote></details>

OK

---

*[openprocurement/chronograph/design.py, line 29 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KtQXGe3658uBbFaYD7z-r4-29:-KuUNSvkRIdsJCHnZ8O-:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/design.py#L29)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

INSIDER_WORKING_DAY_START ніби 09:30?
</blockquote></details>

OK

---

*[openprocurement/chronograph/scheduler.py, line 67 at r3](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r3-67:-KuUNftH-MYA-rmLoG3H:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d699f33173ff7b71b853b267b915d8e754bd11a1/openprocurement/chronograph/scheduler.py#L67)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

Додай classic_auction=True і віддавай лише необхідні дані
</blockquote></details>

OK

---

*[openprocurement/chronograph/scheduler.py, line 78 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r4-78:-KuUNhgByKX5lDr4vMnP:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/scheduler.py#L78)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

ping
</blockquote></details>

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10)*

**Comment by kroman0** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[openprocurement/chronograph/scheduler.py, line 120 at r3](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r3-120:-KuUNng-6aSZCNP3Kf7D:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d699f33173ff7b71b853b267b915d8e754bd11a1/openprocurement/chronograph/scheduler.py#L120)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

480 додай як константу
</blockquote></details>

OK

---

*[openprocurement/chronograph/scheduler.py, line 146 at r3](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r3-146:-KuUNp6P5y_0zZsz1MTs:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d699f33173ff7b71b853b267b915d8e754bd11a1/openprocurement/chronograph/scheduler.py#L146)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

І тут константу з переліком.
</blockquote></details>

OK

---

*[openprocurement/chronograph/scheduler.py, line 80 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r4-80:-KuUNkRVeY3MTcxgRIhg:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/scheduler.py#L80)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

pong
</blockquote></details>

OK

---

*[openprocurement/chronograph/scheduler.py, line 124 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r4-124:-KuUNm4HhP9IYsYZgZQT:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/scheduler.py#L124)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

Тут не потрібно змін.
</blockquote></details>

OK

---

*[openprocurement/chronograph/scheduler.py, line 152 at r5](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r5-152:-KuUNqR3rgEMPC25exNk:b32w0x) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/f8bbd17be759969f8748bb725143fa4f4ddac2bb/openprocurement/chronograph/scheduler.py#L152)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

Яке значення shouldStartAfter для dgfInsider?
@yshalenyk @Leits
</blockquote></details>

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10)*

**Comment by VDigitall** [comment]
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[openprocurement/chronograph/design.py, line 26 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KtQXGe3658uBbFaYD7z-r4-26:-KuXjuxT1FxIM9UQxkOA:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/design.py#L26)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/design.py, line 29 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KtQXGe3658uBbFaYD7z-r4-29:-KuXjzAtNg6ENP-Xa5dm:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/design.py#L29)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/scheduler.py, line 67 at r3](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r3-67:-KuXk-8YyZjEREl_V1YA:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d699f33173ff7b71b853b267b915d8e754bd11a1/openprocurement/chronograph/scheduler.py#L67)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/scheduler.py, line 120 at r3](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r3-120:-KuXk2W9V24Jxnm5Omdl:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d699f33173ff7b71b853b267b915d8e754bd11a1/openprocurement/chronograph/scheduler.py#L120)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/scheduler.py, line 146 at r3](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r3-146:-KuXk3873QFYbLcp2kCN:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d699f33173ff7b71b853b267b915d8e754bd11a1/openprocurement/chronograph/scheduler.py#L146)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/scheduler.py, line 78 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r4-78:-KuXk-kxwe279uU-B2dq:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/scheduler.py#L78)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/scheduler.py, line 80 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r4-80:-KuXk0d2WZXpHCoP46nS:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/scheduler.py#L80)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/scheduler.py, line 124 at r4](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r4-124:-KuXk1p0fqmzF0jO-V-0:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/d36264cd79a321651c1af45ee5609c13f6122740/openprocurement/chronograph/scheduler.py#L124)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---

*[openprocurement/chronograph/scheduler.py, line 152 at r5](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10#-KsxWW4563TQm-h3hGfN-r5-152:-KuXk48Xeu9Dy82XwKIZ:b-896fix) ([raw file](https://github.com/openprocurement/openprocurement.chronograph/blob/f8bbd17be759969f8748bb725143fa4f4ddac2bb/openprocurement/chronograph/scheduler.py#L152)):*
<details><summary><i>Previously, kroman0 (Roman Kozlovskyi) wrote…</i></summary><blockquote>

OK
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openprocurement/openprocurement.chronograph/10)*

**Pull request merged**